sphinx/builders/html/__init__.py

```diff
@@ -468,8 +468,8 @@ def prepare_writing(self, docnames: Set[str]) -> None:
         else:
             self.last_updated = None
 
-        logo = path.basename(self.config.html_logo) if self.config.html_logo else ''
-        favicon = path.basename(self.config.html_favicon) if self.config.html_favicon else ''
+        logo = self.config.html_logo if self.config.html_logo else ''
+        favicon = self.config.html_favicon if self.config.html_favicon else ''
```

Comment:
I feel it's enough to use `self.config.html_favicon or ''` at the assignment to `self.globalcontext` without variable.

Reply:
Good idea! Will do.